webapp/store/views.py

```diff
@@ -1,27 +1,45 @@
-from pprint import pformat
+from flask import Blueprint, render_template
+from webapp.store import models
 
-from flask import Blueprint, abort, render_template
-
-from theblues.charmstore import CharmStore
 
 jaasstore = Blueprint(
   'jaasstore', __name__,
   template_folder='/templates', static_folder='/static')
 
-cs = CharmStore("https://api.jujucharms.com/v5")
-
 
 @jaasstore.route('/store')
 def store():
     return render_template('store/store.html')
 
 
-@jaasstore.route('/<entity_name>')
-def details(entity_name):
-    try:
-        entity = cs.entity(entity_name)
-        entity_formatted = pformat(entity, 1, 120)
-    except Exception:
-        return abort(404, "Entity not found {}".format(entity_name))
+@jaasstore.route('/u/<user_name>/<entity_name>/<series>/<version>')
+def user_details_series_version():
+    raise NotImplementedError()
+
+
+@jaasstore.route('/u/<user_name>/<entity_name>/<series>')
+def user_details_series():
+    raise NotImplementedError()
+
+@jaasstore.route('/u/<user_name>/<entity_name>/')
+def user_details():
+    raise NotImplementedError()
+
+@jaasstore.route('/<charm_or_bundle_name>')
+def details(charm_or_bundle_name):
+    charm_or_bundle = models.get_charm_or_bundle(charm_or_bundle_name)
+
+    if charm_or_bundle['is_charm']:
+        return render_template(
+            'store/charm-details.html',
+            context={'charm': charm_or_bundle}
+        )
+    else:
+        return render_template(
+            'store/bundle-details.html',
+            context={'bundle': charm_or_bundle}
+        )
+
+
+
```

Comment:
I suspect this is too many blank lines for Flake8 to be happy - why don't you get me to show you how to install a decent linter in your editor?